app/src/features/surveys/animals/profile/mortality/AnimalMortalityContainer.tsx

```diff
@@ -0,0 +1,118 @@
+import Box from '@mui/material/Box';
+import Skeleton from '@mui/material/Skeleton';
+import { useAnimalPageContext, useSurveyContext } from 'hooks/useContext';
+import { useCritterbaseApi } from 'hooks/useCritterbaseApi';
+import {
+  IMarkingResponse,
+  IMortalityResponse,
+  IQualitativeMeasurementResponse,
+  IQuantitativeMeasurementResponse
+} from 'interfaces/useCritterApi.interface';
+import { useHistory } from 'react-router';
+import { AnimalMortalityCardContainer } from './components/AnimalMortalityCardContainer';
+import { AnimalMortalityMap } from './components/AnimalMortalityMap';
+import { AnimalMortalityToolbar } from './components/AnimalMortalityToolbar';
+
+export interface IMortalityWithSupplementaryData extends IMortalityResponse {
+  markings: IMarkingResponse[];
+  measurements: { qualitative: IQualitativeMeasurementResponse[]; quantitative: IQuantitativeMeasurementResponse[] };
+}
+
+/**
+ * Container for the animal mortality component within the animal profile page
+ *
+ * @return {*}
+ */
+const AnimalMortalityContainer = () => {
```

Comment:
Does this component need useUnsavedChangesDialog()?
I just noticed we are using a history.push() onSave

Reply:
Not needed in this component. This particular push is just navigating to the create mortality page, which then does leverage the `useUnsavedChangesDialog` already.

Reply:
That being said, its being passed in as a prop, but probably it could just be in the child component directly. Probably just trying to make the child component simpler/dumber.